src/data/store.rs

```diff
@@ -464,8 +281,16 @@ impl Store {
     }
 
     pub fn insert_or_update_provider(&mut self, provider: &Provider) {
+        // Update in memory
         self.chats.insert_or_update_provider(provider);
+        // Update in preferences (persist in disk)
         self.preferences.insert_or_update_provider(provider);
+        // Update in MolyKit (to update the API key used by the client, if needed)
+        if let Some(_bot_repo) = &self.bot_repo {
+            // Because MolyKit does not currently expose an API to update the clients, we'll remove and recreate the entire bot repo
+            // TODO(MolyKit): Find a better way to do this
+            self.bot_repo = None;
+        }
     }
```

Comment:
I think `BotRepo` should be an actual repository-like interface and not a client interface, it might still hold a main
client/multi_client, but the crate user should be able to update the clients (add new ones, update existing ones, remove, etc.)
It would also be helpful if `BotRepo` can expose the bots without re-fetching every time. 

Or at least, I think we should rename `BotRepo` to something else as it does not follow a repository pattern. Even though I know it's just a client or client aggregator, I often found myself trying to use it like a repo.